pkg/syncer/sync.go

```diff
@@ -26,7 +26,11 @@ func (s *Syncer) SyncPendingCharts(names ...string) error {
 	// There might be problems loading all the charts due to missing dependencies,
 	// invalid/wrong charts in the repository, etc. Therefore, let's warn about
 	// them instead of blocking the whole sync.
-	if err := s.loadCharts(names...); err != nil {
+	// If there is a severe error, abortExecution would be true, and we would end the whole sync (with proper error code)
+	abortExecution, err := s.loadCharts(names...)
```

Comment:
do we really need to abort? It seems that the only case now is when the index is empty? If so is that smth we need to check?

Reply:
I am trying to protect the case where no index file is provided or found in the repo and no charts filter is specified. If that is the case, charts-syncer would just say there is nothing to sync. I would like tool to provide feedback about a possible misconfiguration although we can just leave it as it was.

Reply:
I see and I agree with making sure that the user provides one or the other. What I am not sure though is if the code responsible of that guard clause should be here. It looks to me (although I have very limited knowledge on this codebase) that using this boolean as guard clause could be confusing.

Technically it could be used for other things i.e connectivity is gone and we do not want to retry.

I am wondering if there is a more explicit way up in the stack that we could make that check. Couldn't we just check if the index exists || the filter is set in the oci.New function (where we already check if the index exists?)